components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -751,25 +773,32 @@ private boolean isUseLegacyAPIs(int limit) {
     }
 
     /**
-     * Validate whether filter enhancements are enabled and then append the primary domain name in front of the
-     * attribute value to be searched. Finally update the attribute value in Expression node to new attribute value.
+     * Validate whether filter enhancements are enabled and then return primary default domain name as the domain to
+     * be filtered.
      *
      * @param node Expression node
-     * @return Modified attribute value
+     * @return PRIMARY default domainName if property enabled, Null otherwise.
      */
-    private void applyFilterEnhancementsForUsernameInNode(ExpressionNode node) {
+    private String getFilteredDomainName(ExpressionNode node) {
 
         // Set filter values.
         String attributeName = node.getAttributeValue();
         String filterOperation = node.getOperation();
         String attributeValue = node.getValue();
 
-        if (SCIMCommonUtils.isFilteringEnhancementsEnabled())
-            if (SCIMCommonConstants.EQ.equalsIgnoreCase(filterOperation))
+        if (isFilterUsersAndGroupsOnlyFromPrimaryDomainEnabled()) {
+            if (!StringUtils.contains(attributeValue, CarbonConstants.DOMAIN_SEPARATOR)) {
```

Comment:
Shall we combine the two conditions `isFilterUsersAndGroupsOnlyFromPrimaryDomainEnabled()` and `!StringUtils.contains(attributeValue, CarbonConstants.DOMAIN_SEPARATOR)` and use a single `if condition` instead of two to improve readability? If the combined condition seem long and complex, you can wrap it in a private method with a suitable name reflecting the condition it checks.

Reply:
Fixed